### ISSUE TITLE
Ability to ignore folders such as "scans"

### ISSUE THREAD

**Issue by codicusmaximus** [Libresonic/libresonic#455]
Feature request - add the ability in the settings to define a list of folder names / wildcards that should be ignored.  The issue is that when an Album folder contains a folder named "scans" or "extras" it shows up as a clickable item in both the Web UI and most phone clients yet when you click them they contain no music.  It might also be a good idea to simply not show folders if they contain no music.

**Comment by biconou** [comment]
Also interested in that and I can code it.

**Comment by codicusmaximus** [comment]
This feature would still be good to have, but in the meantime i have discovered that folders that start with a . are not shown (at least on linux, haven't verified Windows)

**Comment by biconou** [comment]
@codicusmaximus You should create a separate issue about your latest comment.

**Comment by codicusmaximus** [comment]
That's not an issue, it's a feature.  Folders that start with a . are considered hidden files (by both linux and windows).  This is expected behavior.

**Comment by biconou** [comment]
Indeed. 
All directories whose name is starting with a dot are ignored by this piece of code in `org.libresonic.player.service.MediaFileService`

```
   /**
     * Returns whether the given file is excluded.
     *
     * @param file The child file in question.
     * @return Whether the child file is excluded.
     */
    private boolean isExcluded(File file) {

        // Exclude all hidden files starting with a single "." or "@eaDir" (thumbnail dir created on Synology devices).
        String name = file.getName();
        return (name.startsWith(".") && !name.startsWith("..")) || name.startsWith("@eaDir") || name.equals("Thumbs.db");
    }
```

@codicusmaximus is that feature sufficient for your needs ? You could rename all your scan folders to .scan

**Comment by codicusmaximus** [comment]
@biconou the '.' folder hiding is sufficient for my needs (already working on renaming my folders); however, I still think this feature request is worthwhile and sort of low hanging fruit.

**Comment by biconou** [comment]
@codicusmaximus What is the use of your scans folder ? 
I suppose it is to store some images. In that case it would be cool to be able to display these images in the Libresonic UI while showing the album informations. 
What do you think ?

**Comment by codicusmaximus** [comment]
Yeah that would be neat though significantly more work to implement, so maybe we can open another "enhancement"-tagged issue for that?  Another example is a good number of my albums come with .nsf .sav .it .mod .xm or .s3m original files which were used to render the mp3/flac file.  I place these in a folder named ".originals" since otherwise it would show as a folder which when clicked would be empty.

**Comment by SuperNerd-James** [comment]
@codicusmaximus You actually CAN create files and folders in Windows that start with dots, you just have to use the Command Prompt. Click Start, and use Search for Run, and if using a Classic Shell/Start8 start menu restorer, hit Start, Run, type cmd, hit enter.

You can create a folder using the Command Prompt with (notice the dots?):

mkdir .foldername

You can create a file using command prompt with:

echo.>.filename.extension, or just .filename

Windows' GUI will not allow it, but it can be done. You don't need 3rd party stuff at all.
